Quest storage

OctoOsmo opened the pull request:
Добавил переход на модуль игры по кнопку play
Добавил log4j2
Новый модуль sequrity
Реализована валидация прав на создание и удаление

SoloAs committed:
Now delete buttons available only for admin users.
Authentication lag was fixed

OctoOsmo committed:
resources, users and user roles sql script was added

OctoOsmo committed:
log4j2 support was added

OctoOsmo committed:
button icons was added

OctoOsmo committed:
security module has been created

OctoOsmo committed:
Merge branch 'dev-branch' of https://github.com/Alexander091/text-adventure into quest-storage

Conflicts:
	pom.xml
	text-adventure-app/text-adventure-app-web/pom.xml

	Quest wrapper was added
	File uploader template was added
	Small ui fixes
	wildfly standalone.xml was added into security module

OctoOsmo committed:
system out was replaced by log4j2 in QuestBean

OctoOsmo committed:
QuestDAO interface was set to Local

Alexander091 commented on security/pom.xml:
```diff
@@ -0,0 +1,33 @@
+
+    <artifactId>security</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
```
Этот модуль тоже должен быть разделён на api, impl, ejb и т.д если нужно.

Alexander091 commented on security/src/main/java/org/my/adventure/security/SessionBean.java:
```diff
@@ -0,0 +1,61 @@
+
+    public String getSessionUsername(){
+        String res = "";
+        for (Map.Entry<String, Object> entry : sessionMap.entrySet()) {
```
Что-то как-то странно достаётся имя пользователя

Alexander091 commented on security/src/main/java/org/my/adventure/security/SessionBean.java:
```diff
@@ -0,0 +1,61 @@
+    public boolean isAdmin(){
+        HttpServletRequest request = (HttpServletRequest) FacesContext.getCurrentInstance().getExternalContext().getRequest();
+        String username = request.getRemoteUser();
+        return request.isUserInRole("admin");
```
Не используй литералы. Вынеси в константы

OctoOsmo committed:
deleteById method was added into questDAOimpl
log4j-web dependency was added

OctoOsmo committed:
scope of quest storage impl and security was changed from provided to compile

OctoOsmo committed:
Merge remote-tracking branch 'origin/dev-branch' into quest-storage

Conflicts:
	dao-manager/dao-manager-impl/src/main/resources/hibernate.cfg.xml
	pom.xml
	quest-storage/quest-storage-webapp/pom.xml
	text-adventure-app/text-adventure-app-ear/target/application.xml
	text-adventure-app/text-adventure-app-web/pom.xml

OctoOsmo committed:
standalone-full.xml was added
merged with dev

OctoOsmo committed:
standalone-full.xml was added
merged with dev

OctoOsmo committed:
uploadController was renamed

OctoOsmo committed:
uploadController was renamed

OctoOsmo committed:
Security module was divided into submodules

OctoOsmo committed:
build of security module was fixed
quest-editor still broken

OctoOsmo committed:
fix of quest-editor-webapp build

OctoOsmo committed:
session timeout has been extended

OctoOsmo committed:
Build with Local interface was finaly fixed
Some changes was make in editor and game modules
ejb module was added into quest storage

OctoOsmo committed:
uploader was removed from quest-storage module

OctoOsmo committed:
genre filter has been created
security/session bean has been removed
LoginInfo class has been added to security module

OctoOsmo committed:
Log4j2 was configured

Alexander091 commented on dao-manager/dao-manager-ejb/src/main/java/resources/log4j2.xml:
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
А почему этот файл находится в этом модуле? Нужно на Application уровень его вынести, чтобы другие модули неявно не зависели от dao-manager-ejb и его настроек логирования

Alexander091 commented on quest-editor/quest-editor-webapp/pom.xml:
```diff
@@ -20,7 +20,7 @@
             <artifactId>quest-editor-web</artifactId>
             <version>1.0-SNAPSHOT</version>
-            <type>war</type>
+            <type>jar</type>
```
webapp это war не jar

Alexander091 commented on dao-manager/dao-manager-ejb/src/main/java/org/my/adventure/dao_manager/ejb/dao/QuestDAOImpl.java:
```diff
@@ -17,9 +22,43 @@
 @Local(QuestDAO.class)
 public class QuestDAOImpl extends CommonDAOImpl<Quest> implements QuestDAO{
 
+    private final Logger log = LogManager.getLogger(QuestDAOImpl.class);
```
Логгеры должны быть статичными.
